cirq/protocols/phase.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, TypeVar
+
+from typing_extensions import Protocol
+
+# This is a special value to indicate that the gate should be returned
+# unchanged if underlying implementation of _phase_by_ exists.
+ReturnSelfIfNotProvided = 1.234567
```

Comment:
Use a value that has reference identity, so that you can guarantee it is impossible for a caller to produce the same value. `([],)` is a simple example of such a value. (You don't want to use `[]` because it sets off mutable-default-value alarms, and you don't want to use `()` because it's a singleton. `() is ()` but `([],) is not ([],)`)